### PR TITLE
Pt 164062750 add info field to key header

### PR DESCRIPTION
**Pull request by gorillainduction**
https://www.pivotaltracker.com/story/show/164062750

### CODE REVIEW

**Comment by lucafavatella** [comment]
Auxiliary CI job running all system tests on this PR: https://circleci.com/gh/aeternity/aeternity/8380

File: apps/aecore/src/aec_headers.erl [hunk 1]
```diff
@@ -90,7 +90,8 @@
           version                                              :: non_neg_integer(),
           pow_evidence = no_value                              :: aeminer_pow_cuckoo:solution() | no_value,
           miner        = <<0:?MINER_PUB_BYTES/unit:8>>         :: miner_pubkey(),
-          beneficiary  = <<0:?BENEFICIARY_PUB_BYTES/unit:8>>   :: beneficiary_pubkey()
+          beneficiary  = <<0:?BENEFICIARY_PUB_BYTES/unit:8>>   :: beneficiary_pubkey(),
+          info         = <<>>                                  :: optional_info()
```
Comment (lucafavatella) [inline apps/aecore/src/aec_headers.erl]: I shall check DB compatibility from old DB.

**Comment by lucafavatella** [comment]
I started CI workflows with test in https://github.com/aeternity/aeternity/pull/2181 re reusing Roma DB:
* (Shall fail) On old code in this PR i.e. without the recent two commit addressing DB: https://circleci.com/workflow-run/008b0a66-ffaa-454e-991e-7d79d884f982
* (Shall pass) On latest commit in the PR: https://circleci.com/workflow-run/dca4a63f-5f2d-4703-8e71-9eff5edbdb88

File: config/swagger.yaml [hunk 2]
```diff
@@ -1933,6 +1935,7 @@ definitions:
       - target
       - time
       - version
+      - info
```
Comment (lucafavatella) [inline config/swagger.yaml]: I see same Swagger type used for response of [pending path](https://github.com/aeternity/protocol/blob/0d178475a5794fa7bbc623c4a0f4c163123d580d/node/api/mining_api_usage.md#aeternity-node-modes) so ok.

**Comment by gorillainduction** [comment]
Restart CI after failure of the same type as https://www.pivotaltracker.com/story/show/164073691

File: apps/aecore/src/aec_headers.erl [hunk 3]
```diff
@@ -357,7 +394,9 @@ serialize_for_client(#key_header{} = Header, PrevBlockType) ->
           <<"beneficiary">>   => aehttp_api_encoder:encode(account_pubkey, Header#key_header.beneficiary),
           <<"target">>        => Header#key_header.target,
           <<"time">>          => Header#key_header.time,
-          <<"version">>       => Header#key_header.version},
+          <<"version">>       => Header#key_header.version,
+          <<"info">>          => aehttp_api_encoder:encode(contract_bytearray, Header#key_header.info)
```
Comment (lucafavatella) [inline apps/aecore/src/aec_headers.erl]: This means `cb_` prefix?

Comment (lucafavatella) [reply apps/aecore/src/aec_headers.erl]: I see yes:
https://github.com/aeternity/protocol/pull/315/files#diff-029785cf793779e9ae524bd4d3d3b548R37

**Comment by lucafavatella** [comment]
A system test [failed on CI](https://circleci.com/gh/aeternity/aeternity/8757):
```
%%% aest_sync_SUITE ==> new_node_joins_network: FAILED
%%% aest_sync_SUITE ==> 
Failure/Error: ?assertEqual(#{beneficiary=><<97,107,95,50,86,111,65,104,77,100,55,116,86,74,114,68,89,77,53,118,80,74,119,70,82,106,117,101,90,121,105,114,68,74,117,109,86,74,78,101,66,87,76,57,106,49,101,78,84,72,115,82,120>>,hash=><<107,104,95,50,71,119,55,75,66,116,113,107,57,72,99,118,70,75,69,54,78,83,100,76,112,100,117,87,70,89,100,53,50,74,102,49,114,115,100,83,122,116,82,104,55,98,110,117,50,111,112,75,99>>,height=>30,miner=><<97,107,95,50,74,99,104,102,72,97,50,66,56,49,116,100,122,99,66,88,53,111,99,50,75,120,82,102,81,121,75,87,107,103,101,116,119,107,69,80,88,69,81,120,55,102,106,67,85,112,76,122,85>>,nonce=>3233196453772215092,pow=>[2259,2707,3737,3746,4073,5174,5438,5898,6932,7601,7701,7920,9507,11170,11398,12188,16849,17439,18562,18780,19721,19880,20131,20277,20678,20894,22115,22143,24238,24946,26185,26201,26525,26816,27120,27407,28213,28453,29765,30600,31156,31215],prev_hash=><<107,104,95,50,116,113,84,84,67,122,117,88,112,105,109,86,53,100,82,99,84,66,100,117,81,65,104,69,57,90,86,86,67,122,71,116,74,77,50,116,115,104,101,66,114,113,69,84,75,98,69,76,104>>,prev_key_hash=><<107,104,95,50,116,113,84,84,67,122,117,88,112,105,109,86,53,100,82,99,84,66,100,117,81,65,104,69,57,90,86,86,67,122,71,116,74,77,50,116,115,104,101,66,114,113,69,84,75,98,69,76,104>>,state_hash=><<98,115,95,50,90,53,109,53,82,109,113,87,49,49,53,50,120,112,104,82,119,107,85,120,78,70,70,86,89,122,85,113,117,88,66,121,54,119,75,77,90,65,65,53,83,113,80,89,97,98,69,50,97>>,target=>553692646,time=>1550587814943,version=>1}, Height3)
  expected: #{beneficiary =>
                  <<"ak_2VoAhMd7tVJrDYM5vPJwFRjueZyirDJumVJNeBWL9j1eNTHsRx">>,
              hash =>
                  <<"kh_2Gw7KBtqk9HcvFKE6NSdLpduWFYd52Jf1rsdSztRh7bnu2opKc">>,
              height => 30,
              miner =>
                  <<"ak_2JchfHa2B81tdzcBX5oc2KxRfQyKWkgetwkEPXEQx7fjCUpLzU">>,
              nonce => 3233196453772215092,
              pow =>
                  [2259,2707,3737,3746,4073,5174,5438,5898,6932,7601,7701,
                   7920,9507,11170,11398,12188,16849,17439,18562,18780,19721,
                   19880,20131,20277,20678,20894,22115,22143,24238,24946,
                   26185,26201,26525,26816,27120,27407,28213,28453,29765,
                   30600,31156,31215],
              prev_hash =>
                  <<"kh_2tqTTCzuXpimV5dRcTBduQAhE9ZVVCzGtJM2tsheBrqETKbELh">>,
              prev_key_hash =>
                  <<"kh_2tqTTCzuXpimV5dRcTBduQAhE9ZVVCzGtJM2tsheBrqETKbELh">>,
              state_hash =>
                  <<"bs_2Z5m5RmqW1152xphRwkUxNFFVYzUquXBy6wKMZAA5SqPYabE2a">>,
              target => 553692646,time => 1550587814943,version => 1}
       got: #{beneficiary =>
                  <<"ak_2VoAhMd7tVJrDYM5vPJwFRjueZyirDJumVJNeBWL9j1eNTHsRx">>,
              hash =>
                  <<"kh_2Gw7KBtqk9HcvFKE6NSdLpduWFYd52Jf1rsdSztRh7bnu2opKc">>,
              height => 30,info => <<"cb_Xfbg4g==">>,
              miner =>
                  <<"ak_2JchfHa2B81tdzcBX5oc2KxRfQyKWkgetwkEPXEQx7fjCUpLzU">>,
              nonce => 3233196453772215092,
              pow =>
                  [2259,2707,3737,3746,4073,5174,5438,5898,6932,7601,7701,
                   7920,9507,11170,11398,12188,16849,17439,18562,18780,19721,
                   19880,20131,20277,20678,20894,22115,22143,24238,24946,
                   26185,26201,26525,26816,27120,27407,28213,28453,29765,
                   30600,31156,31215],
              prev_hash =>
                  <<"kh_2tqTTCzuXpimV5dRcTBduQAhE9ZVVCzGtJM2tsheBrqETKbELh">>,
              prev_key_hash =>
                  <<"kh_2tqTTCzuXpimV5dRcTBduQAhE9ZVVCzGtJM2tsheBrqETKbELh">>,
              state_hash =>
                  <<"bs_2Z5m5RmqW1152xphRwkUxNFFVYzUquXBy6wKMZAA5SqPYabE2a">>,
              target => 553692646,time => 1550587814943,version => 1}
      line: 243
.....
```

**Comment by lucafavatella** [comment]
Re state channel test failure in appveyor CI: [tracked](https://www.pivotaltracker.com/n/projects/2159793/stories/164078827).

Re system test failure: is re field info in old version of the node not present in its user API. I will address in distinct PR.

**Pull request merged**